### ISSUE TITLE
Return 'wac-allowed' HTTP header with GET & HEAD responses

### ISSUE THREAD

**Issue by dmitrizagidulin** [solid/node-solid-server#246]
**Problem/Motivation**

There is currently no easy way for solid client code to determine if the current user has read/write/delete etc access to a given resource (short of reading and parsing the ACL chain, which is both discouraged, and not always possible due to ACL limitations on the `.acl` files themselves). (See [solid/#45 - No way to find out if user is authorized to write to a container](https://github.com/solid/solid/issues/45) for context.)

**Proposed Solution**

An easy solution to this would be to add support to LDNode for the following:

With every response, the server should include an additional `WAC-Modes-Allowed:` header in the HTTP response.

For example, doing an unauthenticated HEAD request on a public read-only resource, would yield:

```
WAC-Allowed: user="read", public="read"
```

Doing an HEAD request (while authenticated) to a resource you have full read/write/control access to:

```
WAC-Allowed: user="read;write;control", public=""
```

(Note the `public=""` part -- the empty string denotes no access.)

**Comment by dmitrizagidulin** [comment]
Updated issue to use `Allow:` header instead of CORS `Allow-Methods`

**Comment by rhiaro** [comment]
+1

**Comment by dmitrizagidulin** [comment]
Note that since we don't have a `http://www.w3.org/ns/auth/acl#None` access mode, and since it's problematic to return empty HTTP headers, if the client making the request has no read/write access to a resource, the response will not contain a `WAC-Modes-Allowed` header.

**Comment by dmitrizagidulin** [comment]
Updated it to reflect today's design discussion; return both the permissions of the user, and public permissions.

**Comment by RubenVerborgh** [comment]
Implemented by #550.
